lib/msf/core/post/common.rb

```diff
@@ -105,34 +105,12 @@ def cmd_exec(cmd, args=nil, time_out=15)
       # through /bin/sh, solving all the pesky parsing troubles, without
       # affecting Windows.
       #
-      start = Time.now.to_i
       if args.nil? and cmd =~ /[^a-zA-Z0-9\/._-]/
         args = ""
       end
 
       session.response_timeout = time_out
-      process = session.sys.process.execute(cmd, args, {'Hidden' => true, 'Channelized' => true, 'Subshell' => true })
-      o = ""
-      # Wait up to time_out seconds for the first bytes to arrive
-      while (d = process.channel.read)
-        o << d
-        if d == ""
-          if Time.now.to_i - start < time_out
-            sleep 0.1
-          else
-            break
-          end
-        end
-      end
-      o.chomp! if o
-
-      begin
-        process.channel.close
-      rescue IOError => e
-        # Channel was already closed, but we got the cmd output, so let's soldier on.
-      end
-
-      process.close
+      o = session.sys.process.capture_output(cmd, args, {'Hidden' => true, 'Channelized' => true, 'Subshell' => true })
```

Comment:
params `time_out` is just set to session.response_timeout but not send to function `capture_output` ,as the result `cmd_exec` time_out is always 15.